iep-lwcapi/src/main/scala/com/netflix/iep/lwcapi/EddaObservable.scala

```diff
@@ -0,0 +1,130 @@
+      .headers(headers)
+  }
+
+  case class EddaAutoScalingGroup(name: String,
+                                  start: Long,
+                                  slot: Int)
+  case class EddaInstance(instanceId: String,
+                          vpcId: String,
+                          publicDnsName: Option[String],
+                          publicIpAddress: Option[String],
+                          privateIpAddress: String,
+                          availabilityZone: String,
+                          subnetId: String,
+                          instanceType: String,
+                          lifecycleState: String,
+                          imageId: String,
+                          launchTime: Long,
+                          platform: Option[String],
+                          start: Long,
+                          slot: Long)
+  case class EddaClustersResponse(name: String,
+                                  start: Long,
+                                  autoScalingGroups: List[EddaAutoScalingGroup],
+                                  instances: List[EddaInstance])
+
+  private def parseJson(text: String): Set[Server] = {
+    val response = Json.decode[EddaClustersResponse](text)
+    if (response.instances == null)
+      throw new RuntimeException("instances field in JSON response is missing")
+    response.instances.map { i =>
+      if (i.instanceId == null)
+        throw new RuntimeException("instance instanceId is null")
+      if (i.privateIpAddress == null)
+        throw new RuntimeException("instance privateIpAddress is null")
+      Server(i.instanceId, i.privateIpAddress, 7001)
+    }.toSet
+  }
+
+  def apply(region: String, environment: String, cluster: String): Observable[ServerMessage] = {
+    // request is immutable
+    val request = makeRequest(region, environment, cluster)
+
+    Observable(subscriber => {
+      var previousSet: Set[Server] = Set()
+
+      new Thread(new Runnable() {
+        private def poll(): Set[Server] = {
+          val serverSet: Set[Server] = try {
+            val response = request.asString
+            val contentType = response.header("Content-Type").getOrElse("unset")
+            if (response.code != 200) {
+              logger.warn(s"edda fetch returned http status ${response.code}")
+              previousSet
+            } else if (contentType != "application/json") {
+              logger.warn(s"Response is not application/json: $contentType")
+              previousSet
+            } else {
+              parseJson(response.body)
+            }
+          } catch {
+            case NonFatal(e) =>
+              logger.warn("edda", e) // TODO log properly
+              previousSet
+          }
+
+          previousSet = serverSet
+          previousSet
+        }
+
+        def run(): Unit = {
+          while (!subscriber.isUnsubscribed) {
+            var info = poll()
+            subscriber.onNext(ServerSet(info))
+            Thread.sleep(10000)
```

Comment:
It would seem more idiomatic to me to use something like [Observable.interval](http://reactivex.io/documentation/operators/interval.html) to trigger the refresh.